Lib/warnings.py

```diff
@@ -61,12 +61,13 @@ def _formatwarnmsg_impl(msg):
             tracing = True
             tb = None
         else:
-            tracing = tracemalloc.is_tracing()
             try:
+                tracing = tracemalloc.is_tracing()
                 tb = tracemalloc.get_object_traceback(msg.source)
             except Exception:
                 # When a warning is logged during Python shutdown, tracemalloc
                 # and the import machinery don't work anymore
+                tracing = False
```

Comment:
Shouldn't this be `True`? Like above at line 61. And note below at line 88/89, the "enable tracemalloc" message is appended if tracing is false.

Reply:
I thought that `except Exception` was added because the `tracemalloc` module was optional. In that case it is reasonable that we should not suggest to use the `tracemalloc` module.

 But in fact, it most likely fails because the import system is already broken at Python shutdown, so no new modules can be imported. And if we are in the `else` block, then either the import system works, or `tracemalloc` *was* imported earlier. Can we suggest to enable it? Who knows. I initially thought that we can, running Python with `-X tracemalloc` seems worked in these examples. But I failed to write tests that do not involve running tests via `unittest` (or even `libregrtest`), so I am now not sure that it always works. So perhaps it is safer to set `tracing = True` here.

Reply:
The `tracing` variable is a bit of a misnomer -- it's just used to suppress the "Enable tracemalloc" message. Maybe reverse the sense and rename it to "suggest_tracemalloc"? Then we'd get
```py
# If we're already tracing, don't suggest to enable tracemalloc
suggest_tracemalloc = not tracemalloc.is_tracing()
```
The idea being that if we're already tracing the suggestion is redundant. (See the original PR gh-10486 where it was introduced.)